### PR TITLE
rgw: only bucket owner should be able to remove bucket

### PR DESCRIPTION
**Pull request by yehudasa**
Fixes: http://tracker.ceph.com/issues/18345

Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>

### CODE REVIEW

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -2444,7 +2444,7 @@ void RGWCreateBucket::execute()
 
 int RGWDeleteBucket::verify_permission()
 {
-  if (!verify_bucket_permission(s, RGW_PERM_WRITE)) {
+  if (s->user->user_id.compare(s->bucket_owner.get_id()) != 0)
```
Comment (Jing-Scott) [inline src/rgw/rgw_op.cc]: missing the '{'

**Comment by yehudasa** [comment]
@Jing-Scott fixed, thanks!

**Comment by dmick** [comment]
if you could fix the commit message too before you merge...

**Comment by yehudasa** [comment]
@dmick good point!

**Comment by yehudasa** [comment]
@oritwas can you take a look at this one?

**Comment by yehudasa** [comment]
this should be retargeted to pre-luminous.

**Comment by cbodley** [comment]
@yehudasa looks like there's a conflict with bucket policy here

**Comment by yehudasa** [comment]
@cbodley bucket policy already fixes the original bucket. However, we may want to get the fix into jewel as is.

**Comment by mattbenjamin** [comment]
@Jing-Scott if at all possible, can you rebase?  If not, I will do it

**Comment by cbodley** [comment]
based on @yehudasa's comments above, it sounds like this is no longer needed for master/mimic/luminous

**Pull request closed without merging**